### ISSUE TITLE
Remove dependency on auto_start_flutter

### ISSUE THREAD

**Issue by TheSHEEEP** [gdelataillade/alarm#296]
**Alarm plugin version**
*ex: 4.1.0

**Describe the bug**
The dependency on auto_start_flutter is an issue, since that package is outdated and can no longer be compiled and would need an update. See [this](https://github.com/seena98/auto_start_flutter/issues/18).

There are already a dozen or so forks of that package (including [my own](https://github.com/TheSHEEEP/auto_start_flutter_fixed) now), with apparently different solutions to its compilation issues that work for different people on different versions, etc.
But obviously that is not really a sustainable solution if one wants to use the alarm package for production. The "fixes" I made in my fork, for example, are basically just guesswork.

Maybe the original package will be upgraded, maybe its dev is MIA, who knows. 
Either way, it might be a better idea to integrate this functionality directly in alarm to further reduce dependencies.

**Comment by gdelataillade** [comment]
Hi @TheSHEEEP,

You raise a good point—it might make sense to implement the auto_start_flutter permission check directly within the alarm package. However, I also believe it’s important for a package to remain focused and as simple as possible, staying within its intended scope.

What are your thoughts on this balance?

**Comment by TheSHEEEP** [comment]
I agree, but I'd also say that implementing that check would make alarm simpler to use due to requiring fewer dependencies. The scope or focus wouldn't be changed anyway since the functionality is already there.
I don't think someone could argue that acquiring needed permissions is out of scope for a package.

It happens sometimes that a dependency just becomes unmaintained or disappears, I think what is usually done in that case is to either ~copy-paste~ put the functionality inside the project itself or take up maintenance of the dependency (even if only in the form of an official fork of the original).
Or drop the functionality if the dependency was too much of a burden to maintain - but I don't think that applies here. The auto_start_flutter package works, but its surrounding build files (gradle, manifest, etc.) need a brush up to suit latest Flutter versions.

**Comment by gdelataillade** [comment]
Alright, I'll add this to my to-do list and I'll keep you updated here !

**Comment by orkun1675** [comment]
FWIW we are using https://pub.dev/packages/disable_battery_optimization in a project along with Alarm and it is working well so far.
